Sources/SwifterSwift/Foundation/NSAttributedStringExtensions.swift

```diff
@@ -180,4 +180,22 @@ public extension NSAttributedString {
     }
 }
 
+public extension Array where Element: NSAttributedString {
+    /// SwifterSwift: Returns a new `NSAttributedString` by concatenating the elements of the sequence, adding the given separator between each element.
+    ///
+    /// - Parameter separator: An `NSAttributedString` to add between the elements of the sequence.
+    /// - Returns: NSAttributedString with applied attributes.
+    // https://stackoverflow.com/questions/32830519/is-there-joinwithseparator-for-attributed-strings
+    func joined(separator: NSAttributedString) -> NSAttributedString {
+        guard let firstElt = first else { return NSMutableAttributedString() }
+        return dropFirst().reduce(into: NSMutableAttributedString(attributedString: firstElt)) { result, element in
+            result.append(separator)
+            result.append(element)
+        }
+    }
+
+    func joined(separator: String) -> NSAttributedString {
+        joined(separator: NSAttributedString(string: separator))
```

Comment:
One of the rules of the project is that functions can't depend on one another, so that they can be copy-pasted individually. It leads to code-copying, but it makes life easier as it's often how the project is used. So you'll need to copy the code from the other `joined` function.